### ISSUE TITLE
dot in attribute name

### ISSUE THREAD

**Issue by versatildefuy** [orchestracities/ngsi-timeseries-api#171]
I'm receiving a SQLExpection because some attributes has dot in their names. E.g.
```
    {
        "id": "373773207E330104",
        "type": "AirQualityObserved",
        "PM10": {
            "type": "Number",
            "value": 18.4,
            "metadata": {}
        },
        "PM2.5": {
            "type": "Number",
            "value": 9.5,
            "metadata": {}
        }
    }
```

Can this dot be changed for other symbol before the insertion?

Thank you for your efforts.
Best

**Comment by c0c0n3** [comment]
yes, it can. indeed we were thinking of implementing bidirectional encoding of entity types/attribute names to Crate identifiers---i.e. a bijection `NGSI-Identifier ⟶ Crate-Identifier`. See #136 for more details about similar issues. As discussed there the fix isn't trivial which is why we've postponed it to a later date...

**Comment by versatildefuy** [comment]
Thank you for your prompt response @c0c0n3 . 
I can see you are working on this. 
As you are currently handling the problem in #136 feel free to close this issue.

**Comment by c0c0n3** [comment]
As a temporary workaround, you could introduce a mapping at the agent level so that when data come in from the device the "PM2.5" attribute gets renamed to e.g. "PM2-5" or "PM2_5".

**Comment by taliaga** [comment]
Reopening because I saw latest QL complaining about a dot in the name. I don't recall the full story of what happened with supporting this. We will investigate and check what's the current status.

**Comment by daminichopra** [comment]
Hi @c0c0n3 @taliaga , I have tried reproducing this issue and  it still shows SQLException. Please confirm if it needs to be fixed. Thank you

**Comment by c0c0n3** [comment]
Hi @daminichopra, this one is a hairy issue, like very hairy. The reason why it fails is that the dot isn't a character we can escape, at least not easily---see #136 for the gory details and my [earlier comment](https://github.com/smartsdk/ngsi-timeseries-api/issues/171#issuecomment-470679878) about it.

If this is affecting your deployment, you could try introducing an [agent mapping](https://github.com/smartsdk/ngsi-timeseries-api/issues/171#issuecomment-470860652).
